svr-chansession.c

```diff
@@ -685,28 +685,63 @@ static int sessioncommand(struct Channel *channel, struct ChanSess *chansess,
 		if (chansess->cmd == NULL) {
 			chansess->cmd = buf_getstring(ses.payload, &cmdlen);
 
-			if (cmdlen > MAX_CMD_LEN) {
-				m_free(chansess->cmd);
-				/* TODO - send error - too long ? */
-				return DROPBEAR_FAILURE;
-			}
-			#if DROPBEAR_ONLY_ALLOW_EXEC_SCP
-				#if DROPBEAR_EXEC_REQUEST
-					#define SCP_STRING "scp"
-					if (strncmp(chansess->cmd, SCP_STRING, sizeof(SCP_STRING) - 1) != 0)
-					{
+			#if !DROPBEAR_EXEC_REQUEST && DROPBEAR_ONLY_ALLOW_EXEC_SCP
+				#error "DROPBEAR_EXEC_REQUEST should be enabled if DROPBEAR_ONLY_ALLOW_EXEC_SCP is enabled."
+			#endif
+
+			#if DROPBEAR_EXEC_REQUEST
+				if (cmdlen > MAX_CMD_LEN) {
+					m_free(chansess->cmd);
+					/* TODO - send error - too long ? */
+					return DROPBEAR_FAILURE;
+				}
+
+				#define SCP_STRING "scp"
+				if (strncmp(chansess->cmd, SCP_STRING, sizeof(SCP_STRING) - 1) != 0)
+				{
+					#if DROPBEAR_ONLY_ALLOW_EXEC_SCP
 						#if LOG_COMMANDS
 							dropbear_log(LOG_WARNING, "User %s trying to execute '%s'. Only allowed command is scp. Aborting.",
 												ses.authstate.pw_name, chansess->cmd);
 						#endif
 						m_free(chansess->cmd);
-						return DROPBEAR_FAILURE;
+						return DROPBEAR_FAILURE;	
+					#endif
+				}
+				else
+				{
+					#if DROPBEAR_SCP_FIXED_FILE_PATH_AND_SIZE
+					#define PATH_CHECKING_ARG " -Y "
+					#define MAX_SIZE_CHECKING_ARG " -s "
+					// We build the command again with the neccessary prefixes.
+					char* newcmd = malloc(strlen(svr_opts.allowed_path) + strlen(svr_opts.allowed_max_size) + cmdlen + sizeof(PATH_CHECKING_ARG));
```

Comment:
```+ sizeof(MAX_SIZE_CHECKING_ARG)``` ?

Reply:
check that there is no m_malloc()

Reply:
Done. 
Was not crashing because the sizeof takes into account the \0 and strcat removes the \0 and the sizeof(MAX_SIZE_CHECKING_ARG) is small.